### PR TITLE
Feat/kvm html5 url

### PR DESCRIPTION
**Pull request by varun257**
### Description of the Change

Add new features of sol ssh and kvm html features
Add tiles to group ipmi options
Add ssh sol via ssh key
Change name of jnlp file saved
Add error case for ovhError in task

### Benefits

Adds updated features

### Possible Drawbacks

None

### Applicable Issues

MBP-101, MBP-97, MBP-116

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-manager#242.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✘

**Comment by ovh-cds** [comment]
CDS Report deploy_uxci_2#244.0 ✘  
* deploy_uxci
  * deploy_uxci ✘

File: client/app/dedicated/server/translations/Messages_fr_FR.xml [hunk 1]
```diff
@@ -679,6 +681,15 @@
    <translation id="server_configuration_kvm_access" qtlid="229281">Accéder à mon KVM</translation>
    <translation id="server_configuration_kvm_error" qtlid="229294">Une erreur est survenue lors de la récupération des informations du KVM.</translation>
 
+   <translation id="server_configuration_tile_kvm_heading">Remote KVM</translation>
+   <translation id="server_configuration_tile_kvm_description">Accès au clavier, à la souris et à l'affichage de votre serveur à distance</translation>
+   <translation id="server_configuration_tile_sol_heading">Serial over LAN (SOL)</translation>
+   <translation id="server_configuration_tile_sol_description">Accès à la console série de votre machine via une connexion sécurisée</translation>
+   <translation id="server_configuration_ssh_sol_definition">SSH SOL (via your ssh key)</translation>
+   <translation id="server_configuration_ipmi_actions">Actions</translation>
+   <translation id="server_configuration_ssh_error">Please add ssh key to use this</translation>
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/translations/Messages_fr_FR.xml]: have fixed this

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 2]
```diff
@@ -460,6 +508,55 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+      .catch((data) => {
+        $scope.loader.solSshKeyLoading = false;
+        $scope.loader.buttonStart = false;
+        Alerter.alertFromSWS($translate.instant('server_configuration_impi_navigation_error'), data, $scope.alert);
+      });
+  }
+
+  function getSolSsh() {
+    Server.ipmiGetConnection($stateParams.productId, 'serialOverLanSshKey')
+      .then((data) => {
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: have fixed this

**Comment by varun257** [comment]
@marie-j  Have taken care of all comments and returning promises now. Please have a look

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 3]
```diff
@@ -517,5 +607,17 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
     );
   }
 
+  function loadSshKey() {
+    return Server.getSshKey($stateParams.productId).then(
+      (data) => {
+        $scope.ssh.error = false;
+        $scope.ssh.list = data;
+      },
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: prefer `.catch(() => { ` ... than second function then parameter

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 4]
```diff
@@ -517,5 +607,17 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+      () => {
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: prefer .catch(() => { ... than second function then parameter

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 5]
```diff
@@ -1,212 +1,241 @@
...
-                                  aria-hidden="true">
-                            </span>
-                        </a>
-                    </p>
-                    <ul class="list-inline mt-4">
-                        <li class="text-center"
+            </div>
+            <div class="text-danger"
+                data-ng-if="ipmi.model.activated && (disable.otherTask || disable.localTask)"
+                data-translate="server_configuration_impi_disable">
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: prop alignment ?

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 6]
```diff
@@ -1,212 +1,241 @@
...
-                                             data-ng-if="loader.javaLoading"
+                            <oui-tile-description>
+                                <button type="button"
+                                        class="btn btn-default btn-block"
+                                        data-ng-click="startIpmiNavigation()"
+                                        data-ng-disabled="loader.buttonStart || disable.otherTask || disable.localTask">
+                                    <span data-translate="server_configuration_impi_navigation"></span>
+                                    <oui-spinner class="ml-2"
+                                        data-ng-if="loader.navigationLoading"
+                                        data-size="s">
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: Same (alignment) ?

**Comment by varun257** [comment]
@JeremyDec Have taken care of all comments. Please take a look thanks

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 7]
```diff
@@ -1,212 +1,227 @@
-<div class="row"
-     data-ng-controller="ImpiCtrl"
-     data-ng-init="init()">
+<div
+    class="container-fluid mb-5 px-0"
```
Comment (antleblanc) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: I'm curious about why you are applying `px-0` class to the `.container-fluid` element.

Does the container is helpful in your case?

Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: Its not required. I have removed it

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 8]
```diff
@@ -1,212 +1,227 @@
...
-                    data-ng-disabled="disable.testIpmi || disable.otherTask || disable.localTask">
-                <span>
-                    <span data-translate="server_configuration_impi_test"></span>
-                    <span class="label label-defaut"
-                          data-ng-if="disable.testIpmi"
-                          data-translate="server_configuration_doing">
-                    </span>
-                </span>
-            </button>
+    <div data-ng-if="ipmi.model.activated && disable.testActive">
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: have changed to ul li structure

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 9]
```diff
@@ -1,212 +1,227 @@
...
-                                  data-ng-if="loader.httpDone">
-                            </span>
-                        </span>
-                        <span class="ml-2"
-                              data-ng-class="{ 'font-weight-bold': loader.httpLoading }"
-                              data-translate="server_configuration_impi_http">
-                        </span>
-                    </div>
-                    <div class="mt-2">
+                <div class="col-xm-12 col-md-4">
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]: have removed col-xm-12

**Pull request merged**